src/azure-cli-core/azure/cli/core/parser.py

```diff
@@ -377,26 +400,41 @@ def _check_value(self, action, value):  # pylint: disable=too-many-statements, t
             command_name_inferred = self.prog
             error_msg = None
             if not self.command_source:
-                candidates = difflib.get_close_matches(value, action.choices, cutoff=0.7)
-                if candidates:
-                    # use the most likely candidate to replace the misspelled command
-                    args = self.prog.split() + self._raw_arguments
-                    args_inferred = [item if item != value else candidates[0] for item in args]
-                    command_name_inferred = ' '.join(args_inferred).split('-')[0]
-
+                candidates = []
+                args = self.prog.split() + self._raw_arguments
```

Comment:
Are there any possibilities that `args` begins with an unknown command name, and then `_search_in_extension_commands` would return all the extension names？

Reply:
I don't think so. `_search_in_extension_commands` will return a list of extension names only when the args matches at least one word (not including the beginning `az`) in extension commands. Otherwise it returns `None`.